src/application/services/useEditor.ts

```diff
@@ -0,0 +1,173 @@
+     */
+    onChange: (api: API) => void
+}
+
+/**
+ * Load one tool at a time
+ *
+ * @param src - source path to tool
+ */
+function loadScript(src: string): Promise<Event> {
+  return new Promise(function (resolve, reject) {
+    const script = document.createElement('script');
+
+    script.src = src;
+    script.onload = resolve;
+    script.onerror = reject;
+    document.head.appendChild(script);
+  });
+}
+
+/**
+ * Application service for working with Editor
+ */
+export function useEditor({ content, isReadOnly, onChange }: EditorProps): void {
+  /**
+   * Editor.js instance
+   */
+  const editor = ref<Editor | undefined>(undefined);
+
+  /**
+   * Editor content instance
+   */
+  const refContent = ref<OutputData | undefined>(content);
+
+  /**
+   * User notes tools
+   */
+  const { userEditorTools } = useAppState();
+
+  /**
+   * Function for loading and adding tools to Editor
+   */
+  const mountEditorOnce = async (): Promise<void> => {
+    try {
+      await Promise.allSettled(userEditorTools.value.map((spec: EditorTool) => {
+        if (spec.source.cdn === undefined || spec.source.cdn === '') {
+          return undefined;
+        }
+
+        return loadScript(spec.source.cdn);
+      }));
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const loadedTools: {[key: string]: any } = userEditorTools.value.reduce(
+        (acc, spec: EditorTool) => {
+          // @ts-expect-error: we need to rewrite plugins to TS to get their types
+          const windowPlugin = window[spec.exportName];
+
+          return {
+            ...acc,
+            [spec.title]: windowPlugin,
+          };
+        },
+        {}
+      );
+
+
+      const editorInstance = new Editor({
+        /**
+         * Block Tools
+         */
+        tools: {
+          header: {
+            class: Header,
+            config: {
+              placeholder: 'Title...',
+            },
+          },
+          // image: Image,
+          // code: CodeTool,
+          // list: List,
+          // delimiter: Delimiter,
+          // table: Table,
+          // warning: Warning,
+          // checklist: Checklist,
+          // linkTool: LinkTool,
+          // raw: RawTool,
+          // embed: Embed,
+
+          // /**
+          //  * Inline Tools
+          //  */
+          // inlineCode: InlineCode,
+          // marker: Marker,
+          ...loadedTools,
+        },
+        data: refContent.value,
+        onChange,
+        readOnly: isReadOnly,
+      });
+
+      await editorInstance.isReady;
+
+      editor.value = editorInstance;
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  /**
+   * Update editor instance when user tools was changed
+   */
+  watch(userEditorTools, mountEditorOnce);
+
+  /**
+   * Set editor instance after first mount
+   */
+  onMounted(() => {
+    if (userEditorTools.value.length > 0) {
+      mountEditorOnce().catch(console.info);
```

Comment:
seems like this .catch will never be called since `mountEditorOnce` wrapps the whole body in try-catch